### ISSUE TITLE
Support for shared volume groups

### ISSUE THREAD

**Issue by mersl** [topolvm/topolvm#150]
## What

Use volume groups, which are shared across multiple nodes (multi writing shared disks). So a LV can be accessed on multiple nodes but activated only once. VGs could exist ie on VMware shared disks, ISCSI or FC targets.

Use cases:

- having ONE VG per project with multiple LVs on multiple nodes
- Pacemaker alike clustering for HA

## How

add locking and activation of LVs before the actual mount operation and unlocking/deactivation on POD stop (after unmounting the LV). The magic is, to do this on LV level! 

ie: using clvmd/pacemaker OR lvmlockd/sanlock across all relevant nodes.

This was done some time ago by flex volume plugin (https://github.com/angeloxx/k8s-flexvolume-lvm) already, but has a lot of dependencies on directly accessing LVM etc plus the fact, it got never upgraded to a CSI. This means, the major missing parts are already done on topolvm (ie lvmd ;-) )

Side note: afaik, snapshotting is not supported/allowed on shared VGs/LVs. So snapshot support should be disabled for that type of VGs as of LVM2s limitation.

**Comment by yokaze** [comment]
I think extending `lvmd` to support shared volume groups would not make sense, because such volume groups do not use node topology information. It would be possibly out of scope for TopoLVM.

**Comment by ymmt2005** [comment]
Closing as this seems to be out of scope.
